graph.js

```diff
@@ -1,12 +1,194 @@
 #!/usr/bin/env node
 
-var app = require('commander')
+let app = require('commander')
+let ipfsAPI = require('ipfs-api')
+let jayson = require('jayson')
+let path = require('path')
+let request = require('request')
+const url = require('url')
+const { URL } = url
+
+let Compiler = require('./src/cli/compiler')
+let TypeGenerator = require('./src/cli/type-generator')
+const Logger = require('./src/cli/logger')
+
+// Helper function to construct a subgraph compiler
+function createCompiler(app, subgraphManifest) {
+  // Connect to the IPFS node (if a node address was provided)
+  let ipfs = app.ipfs ? ipfsAPI(app.ipfs) : undefined
+
+  return new Compiler({
+    ipfs,
+    subgraphManifest,
+    outputDir: app.outputDir,
+    outputFormat: app.outputFormat || 'wasm',
+    verbosity: app.verbosity,
+  })
+}
 
 app
   .version('0.1.0')
-  .command('codegen [file]', 'Generates TypeScript types for a subgraph')
-  .command('build [file]', 'Compiles a subgraph and uploads it to IPFS', {
-    isDefault: true,
+  .option(
+    '--verbosity <info|verbose|debug>',
+    'The log level to use (default: LOG_LEVEL or info)',
+    process.env.LOG_LEVEL || 'info'
+  )
+  .option(
+    '-o, --output-dir <path>',
+    'Output directory for build artifacts',
+    path.resolve(process.cwd(), 'dist')
+  )
+  .option('-w, --watch', 'Rebuild automatically when files change')
+  .option('-g, --node <URL>[:PORT]', 'Graph node')
+  .option('-t, --output-format <wasm|wast>', 'Output format (wasm, wast)', 'wasm')
+  .option('-i, --ipfs <addr>', 'IPFS node to use for uploading files')
+  .option('-n, --subgraph-name <NAME>', 'Subgraph name')
+  .option('--api-key <KEY>', 'Graph API key corresponding to the subgraph name')
+
+app.on('--help', function() {
+  console.log('')
+  console.log('Configuration:')
+  console.log('')
+
+  if (app.subgraphName === undefined) {
+    console.log('  Subgraph name: No name defined with -n/--subgraph-name')
+  } else {
+    console.log(`  Subgraph name: ${app.subgraphName}`)
+  }
+
+  if (app.node === undefined) {
+    console.log('  Graph node:    No node defined with -g/--node')
+  } else {
+    console.log(`  Graph node:    ${app.node}`)
+  }
+
+  if (app.ipfs === undefined) {
+    console.log('  IPFS:          No node defined with -i/--ipfs')
+  } else {
+    console.log(`  IPFS:          ${app.ipfs}`)
+  }
```

Comment:
Is this mostly a placeholder for when we have an external way to configure these parameters? It's strange for `--help` to take parameters.

Reply:
I see your point, this could be confusing. I'll change this to only log when there's missing configuration.